### PR TITLE
docs: refactor annotations of extension classes

### PR DESCRIPTION
**Pull request by majadlymhmd**
## What this PR changes/adds

Add missing Extension annotations and replace String with constant NAME
## Why it does that
to make the code more readable and improve code quality

## Linked Issue(s)

ref #2710 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-edc/Connector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [Etiquette for pull requests](https://github.com/eclipse-edc/Connector/blob/main/pr_etiquette.md) for details_)

### CODE REVIEW

File: core/control-plane/contract-core/src/main/java/org/eclipse/edc/connector/contract/ContractNegotiationCommandExtension.java [hunk 1]
```diff
@@ -30,10 +30,12 @@
  * handlers the core provides.
  */
 @CoreExtension
-@Provides({ CommandHandlerRegistry.class })
-@Extension(value = "Contract Negotiation command handler")
+@Provides({CommandHandlerRegistry.class})
+@Extension(ContractNegotiationCommandExtension.NAME)
 public class ContractNegotiationCommandExtension implements ServiceExtension {
 
+    public static final String NAME = "Contract Negotiation command handler";
```
Comment (ndr-brt) [inline core/control-plane/contract-core/src/main/java/org/eclipse/edc/connector/contract/ContractNegotiationCommandExtension.java:37]: please override the `name` method and make it return the `NAME` constant as well

File: core/control-plane/transfer-core/src/main/java/org/eclipse/edc/connector/transfer/TransferProcessCommandExtension.java [hunk 2]
```diff
@@ -30,8 +31,9 @@
  * Registers command handlers that the core provides
  */
 @CoreExtension
+@Extension(TransferProcessCommandExtension.NAME)
 public class TransferProcessCommandExtension implements ServiceExtension {
-
+    public static final String NAME = "Transfer Process Command";
```
Comment (ndr-brt) [inline core/control-plane/transfer-core/src/main/java/org/eclipse/edc/connector/transfer/TransferProcessCommandExtension.java:36]: ditto

File: extensions/common/http/jetty-core/src/main/java/org/eclipse/edc/web/jetty/JettyExtension.java [hunk 3]
```diff
@@ -31,9 +32,10 @@
 import java.security.NoSuchAlgorithmException;
 import java.security.cert.CertificateException;
 
-@Provides({ WebServer.class, JettyService.class })
+@Provides({WebServer.class, JettyService.class})
```
Comment (ndr-brt) [inline extensions/common/http/jetty-core/src/main/java/org/eclipse/edc/web/jetty/JettyExtension.java:35]: The default formatting for static array definitions should have the whitespace after open bracket and before closing bracket

**Review by ndr-brt — COMMENTED** [review]
Please override `name` method where the `NAME` constant is defined, and avoid changing formatting on the `{ }` thing

**Review by ndr-brt — APPROVED** [review]

File: extensions/control-plane/store/sql/contract-negotiation-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtension.java [hunk 4]
```diff
@@ -29,10 +29,12 @@
 
 import java.time.Clock;
 
-@Provides({ ContractNegotiationStore.class })
-@Extension(value = "SQL contract negotiation store")
+@Provides({ContractNegotiationStore.class})
```
Comment (juliapampus) [inline extensions/control-plane/store/sql/contract-negotiation-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractnegotiation/SqlContractNegotiationStoreExtension.java:32]: Whitespace seems to be still missing. See https://github.com/eclipse-edc/Connector/pull/2840#discussion_r1182185925

File: extensions/control-plane/store/sql/contract-definition-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractdefinition/SqlContractDefinitionStoreExtension.java [hunk 5]
```diff
@@ -29,13 +29,15 @@
 import org.eclipse.edc.transaction.datasource.spi.DataSourceRegistry;
 import org.eclipse.edc.transaction.spi.TransactionContext;
 
-@Provides({ ContractDefinitionStore.class })
-@Extension(value = "SQL contract definition store")
+@Provides({ContractDefinitionStore.class})
```
Comment (juliapampus) [inline extensions/control-plane/store/sql/contract-definition-store-sql/src/main/java/org/eclipse/edc/connector/store/sql/contractdefinition/SqlContractDefinitionStoreExtension.java:32]: whitespace

File: extensions/control-plane/store/sql/asset-index-sql/src/main/java/org/eclipse/edc/connector/store/sql/assetindex/SqlAssetIndexServiceExtension.java [hunk 6]
```diff
@@ -28,10 +28,12 @@
 import org.eclipse.edc.transaction.datasource.spi.DataSourceRegistry;
 import org.eclipse.edc.transaction.spi.TransactionContext;
 
-@Provides({ AssetIndex.class, DataAddressResolver.class })
-@Extension(value = "SQL asset index")
+@Provides({AssetIndex.class, DataAddressResolver.class})
```
Comment (juliapampus) [inline extensions/control-plane/store/sql/asset-index-sql/src/main/java/org/eclipse/edc/connector/store/sql/assetindex/SqlAssetIndexServiceExtension.java:31]: whitespace

**Review by juliapampus — CHANGES_REQUESTED** [review]
Rest looks good 👍🏻

**Comment by ndr-brt** [comment]
@majadlymhmd could you resolve conflicts so we can merge?

**Pull request closed without merging**